tests/test_dataset.py

```diff
@@ -1146,3 +1146,37 @@ def test_tiled_tiff_dataset_get_or_create():
         raise Exception(expected_error_msg)
     except AssertionError:
         pass
+
+
+def test_changing_layer_bounding_box():
+    delete_dir("./testoutput/test_changing_layer_bounding_box/")
+    copytree(
+        "./testdata/simple_tiff_dataset/",
+        "./testoutput/test_changing_layer_bounding_box/",
+    )
+
+    ds = TiffDataset("./testoutput/test_changing_layer_bounding_box/")
+    layer = ds.get_layer("color")
+    mag = layer.get_mag("1")
+
+    bbox_offset = ds.properties.data_layers["color"].get_bounding_box_offset()
+    bbox_size = ds.properties.data_layers["color"].get_bounding_box_size()
+    assert bbox_size == (265, 265, 10)
+    original_data = mag.read(bbox_size)
+    assert original_data.shape == (1, 265, 265, 10)
+
+    layer.set_bounding_box(bbox_offset, (100, 100, 10))  # decrease boundingbox
```

Comment:
@philippotto do you prefer one method that can change the offset and the size at the same time or should I split this into two methods?

Reply:
I think, it would be cool to have all three methods :)